integrations/wiz/client.py

```diff
@@ -0,0 +1,240 @@
+from datetime import datetime, timedelta
+from enum import StrEnum
+from typing import Any, AsyncGenerator
+
+import httpx
+from loguru import logger
+
+from port_ocean.context.event import event
+from port_ocean.utils import http_async_client
+
+PAGE_SIZE = 50
+AUTH0_URLS = ["https://auth.wiz.io/oauth/token", "https://auth0.gov.wiz.io/oauth/token"]
+COGNITO_URLS = [
+    "https://auth.app.wiz.io/oauth/token",
+    "https://auth.gov.wiz.io/oauth/token",
+]
+
+ISSUES_GQL = """
+query IssuesTable(
+  $filterBy: IssueFilters
+  $first: Int
+  $after: String
+  $orderBy: IssueOrder
+) {
+  issues: issuesV2(
+    filterBy: $filterBy
+    first: $first
+    after: $after
+    orderBy: $orderBy
+  ) {
+    nodes {
+      id
+      sourceRule {
+        __typename
+        ... on Control {
+          id
+          name
+          controlDescription: description
+          resolutionRecommendation
+          securitySubCategories {
+            title
+            category {
+              name
+              framework {
+                name
+              }
+            }
+          }
+        }
+        ... on CloudEventRule {
+          id
+          name
+          cloudEventRuleDescription: description
+          sourceType
+          type
+        }
+        ... on CloudConfigurationRule {
+          id
+          name
+          cloudConfigurationRuleDescription: description
+          remediationInstructions
+          serviceType
+        }
+      }
+      createdAt
+      updatedAt
+      dueAt
+      type
+      resolvedAt
+      statusChangedAt
+      projects {
+        id
+        name
+        slug
+        businessUnit
+        riskProfile {
+          businessImpact
+        }
+      }
+      status
+      severity
+      entitySnapshot {
+        id
+        type
+        nativeType
+        name
+        status
+        cloudPlatform
+        cloudProviderURL
+        providerId
+        region
+        resourceGroupExternalId
+        subscriptionExternalId
+        subscriptionName
+        subscriptionTags
+        tags
+        createdAt
+        externalId
+      }
+      serviceTickets {
+        externalId
+        name
+        url
+      }
+      notes {
+        createdAt
+        updatedAt
+        text
+        user {
+          name
+          email
+        }
+        serviceAccount {
+          name
+        }
+      }
+    }
+    pageInfo {
+      hasNextPage
+      endCursor
+    }
+  }
+}
+"""
+
+
+class CacheKeys(StrEnum):
+    ISSUES = "wiz_issues"
+    TOKENS = "wiz_tokens"
+
+
+class WizClient:
+    def __init__(
+        self, api_url: str, client_id: str, client_secret: str, token_url: str
+    ):
+        self.api_url = api_url
+        self.client_id = client_id
+        self.client_secret = client_secret
+        self.token_url = token_url
+
+        self.http_client = http_async_client
+        self.http_client.headers.update(self.api_auth_header)
```

Comment:
why are u setting the headers if u r applying it again every request

Reply:
Thank you for catching this. I have removed it.